syntax="proto3";

option java_package = "comm.gameComm";

enum Command {
    GAME_TERMINATION = 0;
    GENERATE_MOVE = 1;
    PLAY_MOVE = 2;
    MATCH_RESET = 3;
    RECV_FAIL = 4;
    CLIENT_DISCONNECT = 5;
    UNKNOWN = 6;
    GET_COMMAND = 7;
    CONNECT = 8;
}

message Move {
    Command command = 1;
    oneof data {
        string player = 2;
        int32 move = 3;
    }
    bool success = 4;
}

message Player {
    string name = 1;
}

service GameComm {
    rpc GameStream(stream Move) returns(stream Move);
}